numpy/_core/tests/test_strings.py

```diff
@@ -859,6 +859,37 @@ def test_rpartition(self, buf, sep, res1, res2, res3, dt):
         assert_array_equal(act3, res3)
         assert_array_equal(act1 + act2 + act3, buf)
 
+    @pytest.mark.parametrize("args", [
+        (None,),
+        (0,),
+        (1,),
+        (3,),
+        (5,),
+        (6,),  # test index past the end
+        (-1,),
+        (-3,),
+        ([3, 4],),
+        ([2, 4],),
+        ([-3, 5],),
+        ([0, -5],),
+        (1, 4),
+        (-3, 5),
+        (None, -1),
+        (0, [4, 2]),
+        ([1, 2], [-1, -2]),
+        (1, 5, 2),
+        (None, None, -1),
+        ([0, 6], [-1, 0], [2, -1]),
+    ])
+    def test_slice(self, args, dt):
+        buf = np.array(["hello", "world"], dtype=dt)
+        act = np.strings.slice(buf, *args)
+        bcast_args = tuple(np.broadcast_to(arg, buf.shape) for arg in args)
+        res = np.array([s[slice(*arg)]
+                        for s, arg in zip(buf, zip(*bcast_args))],
+                       dtype=dt)
+        assert_array_equal(act, res)
+
```

Comment:
Would be good to test failures too (i.e., wrong input).

Reply:
I added two simple wrong input examples. Let me know if I should add more.